Model.cpp

```diff
@@ -129,6 +137,15 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
                         conn.setString(paramNum, i->getValue());
                     }
                 }
+                else if(f->getType() == BOOLEAN)
```

Comment:
For a start, the null check could be done before this if statement, which would mean we wouldn't have to repeat it within each block.